src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (args) => {
+    if (args === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    } else {
+      sum += args;
+
+      return adder;
+    }
```

Comment:
Redundant else statement here. You have a return before
```suggestion
      return result;
    } 
    
    sum += args;

    return adder;
```